src/Component.js

```diff
@@ -0,0 +1,67 @@
+/* Represents a component of a vector */
+class Component
+{
+	constructor(length, maxRep, jump, hop, promote)
+	{
+		this.length = length;
+		this.maxRep = maxRep;
+		this.jump = jump;
+		this.hop = hop;
+		this.promote = promote;
+	}
+	
+	static Create(compStr, globalStr)
+	{
+		var comp = new Component(0, 1, false, false, false);
+		comp.length = Number(compStr.match(/(-?\d+)/g)[0]);
+		
+		/* Check for jump on first component or end */
+		if (compStr.includes("j") || globalStr.includes("j"))
+		{
+			comp.jump = true;
+		}
+		
+		/* Check for hop on first component or end */
+		if (compStr.includes("h") || globalStr.includes("h"))
+		{
+			comp.hop = true;
+		}
+		
+		/* Check for promote on first component or end */
+		if (compStr.includes("p") || globalStr.includes("p"))
+		{
+			comp.promote = true;
+		}
+		
+		/* Check for finite repetition on end */
+		var repetition = globalStr.match(/{(\d+)}/g);
```

Comment:
Also use `const` here. Since the comment is here, you could also move the `match` call into the actual condition. The expression is simple enough and is used only once.

Reply:
Cannot remove the variable declaration entirely because it is used later (in that if statement).